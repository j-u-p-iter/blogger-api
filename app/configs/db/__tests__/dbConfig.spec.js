import createDBConfig from '../dbConfig';


describe('createDBConfig(environmentObject)', () => {
  describe('when DB_PORT, DB_HOST and DB_NAME  was defined', () => {
    it('should return correct config', () => {
      const DB_PORT = 2000;
      const DB_HOST = 'some-host';
      const DB_NAME = 'some-db-name';
      const environmentObject = {
        DB_PORT,
        DB_HOST,
        DB_NAME,
      };

      const result = createDBConfig(environmentObject);

      expect(result).toMatchSnapshot();
    });
  });

  describe('when DB_PORT was not defined', () => {
    it('should throw', () => {
      const environmentObject = {
        DB_HOST: 'some-host',
      };

      expect(createDBConfig.bind(null, environmentObject))
        .toThrowError(/DB config validation error/);
    });
  });

  describe('when DB_HOST was not defined', () => {
    it('should throw', () => {
      const environmentObject = {
        DB_PORT: 12345,
      };

      expect(createDBConfig.bind(null, environmentObject))
        .toThrowError(/DB config validation error/);
    });
  });

  describe('when DB_NAME was not defined', () => {
    it('should throw', () => {
      const environmentObject = {
        DB_PORT: 12345,
        DB_HOST: 'some-host',
      };

      expect(createDBConfig.bind(null, environmentObject))
        .toThrowError(/DB config validation error/);
    });
  });

  describe('when DB_PORT is not a number', () => {
    it('should throw', () => {
      const environmentObject = {
        DB_PORT: 'not-a-number',
        DB_HOST: 'some-host',
      };

      expect(createDBConfig.bind(null, environmentObject))
        .toThrowError(/DB config validation error/);
    });
  });

  describe('when DB_HOST is not a string', () => {
    it('should throw', () => {
      const environmentObject = {
        DB_PORT: 5000,
        DB_HOST: 12345,
      };

      expect(createDBConfig.bind(null, environmentObject))
        .toThrowError(/DB config validation error/);
    });
  });

  describe('when DB_NAME is not a string', () => {
    it('should throw', () => {
      const environmentObject = {
        DB_PORT: 12345,
        DB_HOST: 'some-host',
        DB_NAME: 12345,
      };

      expect(createDBConfig.bind(null, environmentObject))
        .toThrowError(/DB config validation error/);
    });
  });

  describe('when DB_NAME length < 5', () => {
    it('should throw', () => {
      const environmentObject = {
        DB_PORT: 12345,
        DB_HOST: 'some-host',
        DB_NAME: '<5',
      };

      expect(createDBConfig.bind(null, environmentObject))
        .toThrowError(/DB config validation error/);
    });
  });
});
