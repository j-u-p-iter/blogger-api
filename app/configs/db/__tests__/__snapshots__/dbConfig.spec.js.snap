// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`createDBConfig(environmentObject) when DB_PORT, DB_HOST and DB_NAME  was defined should return correct config 1`] = `
Object {
  "DB_HOST": "some-host",
  "DB_NAME": "some-db-name",
  "DB_PORT": 2000,
}
`;
